packages/block-editor/src/components/iframe/use-scale-canvas.js

```diff
@@ -456,7 +483,7 @@ export function useScaleCanvas( {
 	return {
 		isZoomedOut,
 		scaleContainerWidth,
-		contentResizeListener,
-		containerResizeListener,
+		contentRefCallback,
+		containerRefCallback,
```

Comment:
Let's call it just `contentRef`/`containerRef`. The fact that it's a callback is an implementation detail that nobody really needs to know.

Also I think React Compiler treats variables differently when their name has the `*Ref` suffix.

Reply:
> Let's call it just `contentRef`/`containerRef`. The fact that it's a callback is an implementation detail that nobody really needs to know.

When I'm writing my own React code, I typically differentiate between `RefObject<T>` and `RefCallback<T>` with different suffixes so you know if you can access a `.current` property or not.

I don't see anyone else doing that in Gutenberg though, so I suppose I can change it here.

> Also I think React Compiler treats variables differently when their name has the `*Ref` suffix.

Thanks for bringing that to my attention. It looks like React Compiler bases it on the `useRef()` call, not the name of the variable.

https://playground.react.dev/#N4Igzg9grgTgxgUxALhAgHgBwjALgAgBMEAzAQygBsCSoA7OXASwjvyjAQFEA3BO3AAo4ZSpQBGZOAGsAlPmAAdNvjiswBGKXwBedpwBKpQYPk6AfAuX4b+XAAsYEAO746CV1xhOYg67ZsAcgBhUQkpaTBVLTJcBEJ8ZyYHfAADDm4+AVTVMjo6CAJxBFyxeKJYJjoAc3wtOmIYKuqAOkD-G1kAbn8AX27-DIBJOk48FjouEhIERhMzSyUVGy0SFrhYeoI9ETFJGR6VfsOVhFxYNgzQvYi-ZfxBFqeyGGqwBbrSdc3+XAB+FqPZ6vd4AGg6+AA2gBdfwDOi9ZTKDDYPBEUgUaj4WgMZisfAAWQAngBBTCYUxWFRqUYEAC2RKMJF0+kyv3mukW+GO-hpGnwDKZ13CMhZGV47MpFgU3Nk-i05xgbDuAQAPOYITZVYQmDxPiQdMBBaRevgAPQa+5anV61aG40kYX7aSmi0Q1VulTwxF0EC9IA

I know ESLint's `react/compiler` rules are still a little wonky. Maybe ESLint was using the `*Ref` suffix for checking before, but I don't see any errors now. 🤷

Reply:
> It looks like React Compiler bases it on the `useRef()` call

@tyxla had to change a lot of ref variable names in #64718 to prevent a "mutating a value" error from the compiler. So the names matter.

In this case `contentRefCallback` would probably still be fine, because there is never any `.current = ...` assignment related to `contentRefCallback`.

Reply:
The ESLint plugin has been enabled since https://github.com/WordPress/gutenberg/pull/61788, so it should automatically detect if we have a problem with the ref and if it needs to be named with `*Ref`. If it didn't detect it, then you're most likely not mutating the ref's `.current`, which is totally fine.

Reply:
I did some more testing in the React Compiler Playground. I didn't have the	`enableTreatRefLikeIdentifiersAsRefs` option set which corresponds to the ESLint rule that you're talking about. The new example has it enabled.

https://playground.react.dev/#N4Igzg9grgTgxgUxALhAejQAgAIIHYCGARgDYIAqMCBALgEoIBmAMgJYDWCAkgCb42tGrBDDABBMA0ZhMAXkw0YUBAG4AOng0Y5s3Xv0HDR4ycNasAOQIBbBDIiNMVAOZQSBGJihgEUp0xkeCDs8AHIaTB8EawUITGtaGhEAOnMdUwzMrP0NDUYoPDgBCDxMAFkATykACgBKTGANTEw4ErAI6yqmOS8fGrw3Elr1Ut6EAFFGRgQi6rq5AD4Gpub4rsZkuFgqPAj5AZISEeaAX2GVqhpYUoAeHlYAN39GWWBOqRPMNAWRk9y8fKFYqlSoAeSIACsZjR5o1Rq08O01uCoUUet5fExqgchscxpNprN5rIlnDVl80CtmtoAGIEEg+TAABwgYFYAgeCCpFO52nGAGU2LtkOUoDRaKw8M5MARMA96cp-FcYHgEDxMIwYBAYrLAUVWCVMAB3AAWrIQSuucoVFrAZrc6rwEAiRAt1jFtDVqVGzU6KOhm22-B6OLxZzxlytd0ez1efsh0M+31+-202XTGaMaQYytV6qo0hlO3CMsOECNatimFd8Q9SR43rTmebGf+euBYxz1xqkvZrHp9TJFJabQiBfRfSxvYEA7x2kjKueEYQuc7U7wfdnGj+mgBBX1hsqXZVUhqg5WCKR7xX1zVfnkGOPeH6g3OowxBOhc3qJOWPrWT53kwgYwDseyYKGKzhhcN6LtGTwFnGXS5kBjBJj825tvuHZHrBz5MP6sznvCo4AXhaqEeBj54S+hxvs0H5TF+xKkryWACkKNAimUdaStKsryiQioLnmGpajqGrYQapSmualqLoJip2tAJCOs61ZunWXpsZgXB4IJrA8AwBBFNxvFSjK1pCRaImVpq2qWe20nGmajIiVZSn2qpEHqTW7rivW3rkto14oTwlEgWBIaDGG9Hybc9wIUwSGAeFCZFOhKa7k2La5Rk2bkeqcD0iQRAmewMgmgQnI+S0Qa7MyWpMiINAVFWJmIGAMgEHgFRGgQFSNlgeUjSYWFAs5H6crs1TFYcZVwOwxHNJeY7dA+k6MN+iwNJg0Hvj4ek+DAwKfkSP6sf+BaRcG8hzaV5Wxcuq4YgAwiVC3sHMyQ-R4zhgBdzw3bsAD8yTVD9yR-QDAA0mAANoALpvjueRSYeFTjNN9BYstI6Ih0mPY-e+LY9tv7AHtcWMYSMIsX+wWUv+tL0oyqrOBKnJsdynRY-wUjA+BkGjPtzTufBsZvET-NMJlmG7k5GN87s73zeVsIXqRvPY6rD2LROExk-TlOi-iTHnTtQ7UuxgqSlxooBXxlmKTZhViQ5urozJrmu6uLuRF5akuppAXaczWB6QZRnUKZDsShZAk2vFdniY5XsuXJ7n+8pDo+cHtahw2Ona-wuufYL0WHE9MGrhLiFS8rNBl+Vct4KjeA5aNXe6GkACqbIJ4cCgmharTWCyqq7DIkDDwgbUeBaJX+CZVWkAgQ3pN3Xf-AgAAeLInZgfCMAQbgRIrIIVGITJMhrozudU3I3As3LNDcuGON8r+YO-FSUV8L9-xv1wrmU83Qv5AJ-iA7sBF0oRAgeSYB0tdh+AQYg3+jdm76zQW-NBKMNAgBOEAA

I'm wondering if we should actually have a different name for `RefCallback<T>` variables because they don't have a `.current` property like the `RefObject<T>` variables? A `Ref` suffix makes the ESLint (and React Compiler) think that they do with the `enableTreatRefLikeIdentifiersAsRefs` option enabled.

This PR probably isn't the best place to discuss, so I'll open a discussion and link it here.